langchain4j-azure-ai-search/src/main/java/dev/langchain4j/store/embedding/azure/search/AzureAiSearchEmbeddingStore.java

```diff
@@ -311,17 +402,32 @@ private float[] doublesListToFloatArray(List<Double> doubles) {
 
     /**
      * Calculates LangChain4j's RelevanceScore from Azure AI Search's score.
-     *
-     * Score in Azure AI Search is transformed into a cosine similarity as described here:
-     * https://learn.microsoft.com/en-us/azure/search/vector-search-ranking#scores-in-a-vector-search-results
-     *
-     * RelevanceScore in LangChain4j is a derivative of cosine similarity,
-     * but it compresses it into 0..1 range (instead of -1..1) for ease of use.
      */
-    private double fromAzureScoreToRelevanceScore(double score) {
-        double cosineDistance = (1 - score) / score;
-        double cosineSimilarity = -cosineDistance + 1;
-        return RelevanceScore.fromCosineSimilarity(cosineSimilarity);
+    private double fromAzureScoreToRelevanceScore(double score, QueryType queryType) {
+        if (queryType == QueryType.SIMILARITY) {
+            // Calculates LangChain4j's RelevanceScore from Azure AI Search's score.
+
+           //  Score in Azure AI Search is transformed into a cosine similarity as described here:
+           // https://learn.microsoft.com/en-us/azure/search/vector-search-ranking#scores-in-a-vector-search-results
+
+           // RelevanceScore in LangChain4j is a derivative of cosine similarity,
+           // but it compresses it into 0..1 range (instead of -1..1) for ease of use.
+
+            double cosineDistance = (1 - score) / score;
+            double cosineSimilarity = -cosineDistance + 1;
+            return RelevanceScore.fromCosineSimilarity(cosineSimilarity);
+        } else if (queryType == QueryType.FULL_TEXT) {
+            // Search score is into 0..1 range already
+            return score;
+        } else if (queryType == QueryType.SIMILARITY_HYBRID) {
+            // Search score is into 0..1 range already
+            return score;
+        } else if (queryType == QueryType.SEMANTIC_HYBRID) {
+            // Re-ranker score is into 0..4 range
+            return score/4.0;
```

Comment:
You need to actually pull out the reranker score - it's a separate result
https://azuresdkdocs.blob.core.windows.net/$web/java/azure-search-documents/11.6.2/com/azure/search/documents/models/SearchResult.html#getSemanticSearch()

https://azuresdkdocs.blob.core.windows.net/$web/java/azure-search-documents/11.6.2/com/azure/search/documents/models/SemanticSearchResult.html#getRerankerScore()

Reply:
Thank you so much @mattgotteiner ! I just implemented this, and added more tests to be extra sure.